### PR TITLE
gpg and sigstore sign publications

### PR DESCRIPTION
**Pull request by loosebazooka**
fyi @vlsi I removed the stage/vote plugin because it seemed a bit heavy weight for what I was looking for. Just simple signing.

In my tests, also these setup signing .sigstore with PGP. I'm not sure what I'm misisng.

### CODE REVIEW

File: build-logic/publishing/build.gradle.kts [hunk 1]
```diff
@@ -10,5 +10,6 @@ dependencies {
     implementation(project(":basics"))
     implementation(project(":jvm"))
     implementation("dev.sigstore.build-logic:gradle-plugin")
+    implementation("dev.sigstore:sigstore-gradle-sign-plugin:0.5.0")
```
Comment (vlsi) [inline build-logic/publishing/build.gradle.kts:13]:
```suggestion
    implementation("dev.sigstore.sign:dev.sigstore.sign.gradle.plugin:0.5.0")
```

This way the artifact name and group matches the plugin id

File: build.gradle.kts [hunk 2]
```diff
@@ -1,35 +1,5 @@
...
-
-        Git SHA: ${it.gitSha}
-        Staging repository: ${it.nexusRepositoryUri}
-        """.trimIndent()
-    }
-}
-
-allprojects {
-    version = rootProject.version
-}
```
Comment (vlsi) [inline build.gradle.kts:35]: This was conditionally adding `-SNAPSHOT` when building non-release version. I believe we would like to keep building `-SNAPSHOT` versions by default.

Comment (loosebazooka) [reply build.gradle.kts:35]: ah gotcha. I can fix that.

**Comment by vlsi** [comment]
> Another workaround could be removing the unwanted signatures by calling afterEvaluate { Sign.signatures.remove(...) }

We might factor and ship this as a helper plugin so the users could produce .sigstore without .asc as well.

**Comment by loosebazooka** [comment]
unfortunately the artifacts don't all to seem be available in the afterEvaluate block. I can only modify the .module and .pom files?

**Comment by loosebazooka** [comment]
I'm also not sure how to convert PublicationArtifact to PublishArtifact

**Comment by loosebazooka** [comment]
@vlsi, how about we merge this for now and work on the ignoring .sigstore files in an update?

**Review by vlsi — APPROVED** [review]

**Pull request merged**